File: 2014/Round_1b/A_The_Repeater/TheRepeater.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

struct charInfo {
    char c;
    int length;
};

struct stringInfo {
    struct charInfo chars[101];
    int length;
};

int compareInt(const void *a, const void *b)
{
    return (*(const int *)a - *(const int *)b);
}

void solve(int t)
{
    int N;
    scanf("%d", &N);

    struct stringInfo strings[100] = { 0 };
    char string[101];
    for (int n = 0; n < N; n++) {
        scanf("%s", string);

        char *start = string;
        int length = 0;
        while (start[length]) {
            length++;
            if (start[0] != start[length]) {
                strings[n].chars[strings[n].length].c = start[0];
                strings[n].chars[strings[n].length].length = length;
                strings[n].length++;
                start += length;
                length = 0;
            }
        }
    }

    int match = 1;
    for (int n = 1; n < N && match; n++) {
        if (strings[n - 1].length != strings[n].length) {
            match = 0;
            break;
        }

        for (int i = 0; i < strings[n].length; i++) {
            if (strings[n - 1].chars[i].c != strings[n].chars[i].c) {
                match = 0;
                break;
            }
        }
    }

    if (!match) {
        printf("Case #%d: Fegla Won\n", t + 1);
        return;
    }

    int moves = 0;
    int lengths[100];
    for (int i = 0; i < strings[0].length; i++) {
        for (int n = 0; n < N; n++) {
            lengths[n] = strings[n].chars[i].length;
        }

        qsort(lengths, N, sizeof(int), compareInt);

        int bestLength = lengths[N / 2];
        for (int n = 0; n < N; n++) {
            moves += abs(lengths[n] - bestLength);
        }
    }

    printf("Case #%d: %d\n", t + 1, moves);
}

int main(int argc, char *argv[])
{
    int T;
    scanf("%d", &T);

    for (int t = 0; t < T; t++) {
        solve(t);
    }

    return 0;
}

File: 2014/Round_1b/A_The_Repeater/TheRepeater.py
import sys

def debug(x):
#    print(x)
    return

def solve(t):
    N = int(input())

    strings = [];
    for n in range(N):
        string = input()

        start = string[0]
        length = 0
        chars = []
        for i in range(1, len(string)):
            length = length + 1
            if start != string[i]:
                chars.append((start, length))
                start = string[i]
                length = 0
        chars.append((start, length + 1))
        strings.append(chars)

    for i in range(1, len(strings)):
        if len(strings[i - 1]) != len(strings[i]):
            print(("Case #%d: Fegla Won" % (t + 1)))
            return

        for j in range(len(strings[i])):
            if strings[i - 1][j][0] != strings[i][j][0]:
                print(("Case #%d: Fegla Won" % (t + 1)))
                return

    moves = 0
    for i in range(len(strings[0])):
        lengths = []
        for j in range(len(strings)):
            lengths.append(strings[j][i][1])

        lengths.sort()

        bestLength = lengths[int(len(lengths) / 2)]
        for j in range(len(lengths)):
            moves = moves + abs(lengths[j] - bestLength)

    print("Case #%d: %d" % (t + 1, moves))

T = int(input())

for t in range(T):
    solve(t)